xrpl4j-model/src/test/java/org/xrpl/xrpl4j/model/transactions/TransferFeeTest.java

```diff
@@ -0,0 +1,21 @@
+package org.xrpl.xrpl4j.model.transactions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.google.common.primitives.UnsignedInteger;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigDecimal;
+
+public class TransferFeeTest {
```

Comment:
We should add a test for the `ofPercent` constructor that tries to construct using a `BigDecimal` with more than 2 decimal places (like 50.001) and ensure that an exception is thrown.